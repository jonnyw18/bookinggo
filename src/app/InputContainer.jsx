import React, { Component } from 'react';

export default class InputContainer extends Component {
    constructor(props) {
        super(props)
        this.state = {}
    }

    render() {
        return (
            <h1>Input container</h1>
        )
    }
}