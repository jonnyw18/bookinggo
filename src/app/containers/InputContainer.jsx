import React, { Component } from 'react';
import axios from 'axios';

// Presentational component
import { Input } from '../components/';

export default class InputContainer extends Component {
    constructor(props) {
        super(props)
        this.state = {
            searchData: [],
            userInput: ''
        }
    }

    handleChange = (event) => {
       this.setState({
           userInput: event.target.value
       }, () => {
           this.getLocation()
       })
    }

    getLocation = () => {
        axios.get(`https://cors.io/?https://www.rentalcars.com/FTSAutocomplete.do?solrIndex=fts_en&solrRows=${6}&solrTerm=${this.state.userInput}`)
            .then((response) => {
                if(response.status === 200) {
                    this.setState({
                        searchData: response.data.results.docs
                    })
                }
            }).catch((err) => console.log(err));
    }

    render() {
        return (
            <Input
                searchData={this.state.searchData}
                handleChange={this.handleChange}
            />
        )
    }
}