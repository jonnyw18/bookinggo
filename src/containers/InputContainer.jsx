import React, { Component } from 'react';
import axios from 'axios';

// Presentational components
import { Input, SearchResults } from '../components';

export default class InputContainer extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchData: [],
            userInput: '',
            loading: false,
            showResults: false
        }
    }
    handleChange = (event) => {
       this.setState({
           userInput: event.target.value,
           loading: event.target.value.length > this.state.userInput.length
       }, () => {
           this.getLocation()
       })
    };

    getLocation = () => {
        axios.get(`https://cors.io/?https://www.rentalcars.com/FTSAutocomplete.do?solrIndex=fts_en&solrRows=${6}&solrTerm=${this.state.userInput}`)
            .then((response) => {
                if(response.status === 200) {
                    !response.data.results
                        ? this.setState({ searchData: [], loading: false })
                        : this.setState({ searchData: response.data.results.docs, loading: false })
                }
            }).catch((err) => console.log(err));
    };

    handleOnFocus = () => {
        this.setState({
            showResults: false
        })
    };

    handleOnBlur = () => {
        // this.setState({
        //     showResults: true,
        //     loading: false
        // })
    };

    render() {
        const {
            userInput,
            searchData,
            loading,
            showResults
        } = this.state;
        return (
            <React.Fragment>
                <Input
                    handleChange={this.handleChange}
                    handleOnFocus={this.handleOnFocus}
                    handleOnBlur={this.handleOnBlur}
                    userInput={userInput}
                    loading={loading}
                />
                <SearchResults
                    searchData={userInput.length > 1 ? searchData : []}
                    showResults={showResults}
                />
            </React.Fragment>
        )
    }
}