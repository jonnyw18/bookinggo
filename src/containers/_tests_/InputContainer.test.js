import InputContainer from '../InputContainer';

describe('<InputContainer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InputContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
