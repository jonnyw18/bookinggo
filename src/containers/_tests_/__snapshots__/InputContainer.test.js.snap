// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputContainer /> renders correctly 1`] = `
<React.Fragment>
  <Input
    handleChange={[Function]}
    handleOnBlur={[Function]}
    handleOnFocus={[Function]}
    loading={true}
    selectedLocation=""
    userInput=""
  />
  <SearchResults
    hideResults={false}
    searchData={Array []}
  />
</React.Fragment>
`;
